x-pack/plugins/dataset_quality/common/data_streams_stats/data_stream_stat.ts

```diff
@@ -32,10 +34,13 @@ export class DataStreamStat {
   }
 
   public static create(dataStreamStat: DataStreamStatType) {
-    const [_type, dataset, namespace] = dataStreamStat.name.split('-');
+    const [type, ...dataStreamParts] = dataStreamStat.name.split('-');
+    const namespace = dataStreamParts[dataStreamParts.length - 1];
+    const dataset = dataStreamParts.slice(0, dataStreamParts.length - 1).join('-');
```

Comment:
could we extract this (extracting type, datasetname and namespace) in an util and include some tests?